### ISSUE TITLE
[Question] HELP!!! Parallel environments do not support independent multi-run execution

### ISSUE THREAD

**Issue by ItsBean** [isaac-sim/IsaacLab#973]
### Question

I am working with parallel environments in Isaac Sim and encountered an issue where the state of one environment affects the others during execution. Specifically, when one environment reaches a `done` or `truncated` state, I attempt to reset only that environment, but this results in unintended behavior where the remaining environments are impacted.

Here is the simplified code structure I'm using:
```python
num_envs=10
for _ in range(200):
    step += 1
    action = env.action_space.sample()  # Assuming this returns a (10, action_dim) tensor

    actions = torch.tensor(action, dtype=torch.float32).to(device)  # Shape: (10, 4)

    # Step the environment with all 10 actions
    obs, rewards, dones, truncateds, infos = env.step(actions)
    obs = obs['policy']  # Shape: (num_envs, obs_dim)

    # Reset environments that are done or truncated
    for i in range(num_envs):
        if dones[i] or truncateds[i]:
            new_obs, _ = env.reset(env_id=i)
            obs[i] = new_obs['policy']
```

`env.reset(env_id=i)` is not supported. So Isaac Sim currently doesn't support the functionality where environments can run independently without affecting each other. When one environment finishes (reaches done or truncated), it is not possible to reset only that specific environment without resetting or influencing the others.

### Expectation
I would like a mechanism where **each environment can be managed and reset independently**, such that the progression of one environment does not impact the others.

**Comment by pascal-roth** [comment]
Hi, 
Each environment is reset automatically depending on the event functions you passed into the configuration. Looking into the step function, it is visible that all the environments are already getting reset when they terminate or run out of time. So your reset call is not necessary. The returned observations are already from the resetted environments. 

We are using and testing this behavior and cannot see any effects on other environments. Can you please detail and provide a minimal script where you observe this behavior?

**Comment by pascal-roth** [comment]
the `SequentialTrainer` is outside of the scope of IsaacLab, please raise this issue there.

In our environment definition, each environment is reset individually as given [here](https://github.com/isaac-sim/IsaacLab/blob/8fcbf85f2519389d88e6c9447f4b4836f1143a84/source/extensions/omni.isaac.lab/omni/isaac/lab/envs/manager_based_rl_env.py#L196) whenever it is terminated or truncated. 

To run multiple environments, just increase the number of envs [here](https://github.com/isaac-sim/IsaacLab/blob/190fa8546a9bc743c3f82b8e70aa40f5b2f694af/source/extensions/omni.isaac.lab/omni/isaac/lab/scene/interactive_scene_cfg.py#L70)

**Comment by pascal-roth** [comment]
Yes, exactly; we handle the reset internally.

Not sure why skrl would reset all environments once one terminates. I would recommend using rsl_rl, which does not include this behavior.

**Comment by Toni-SM** [comment]
Hi @ItsBean (and @pascal-roth)

Indeed, Isaac Lab (as well as other frameworks that implement parallel/vectorized environments, such as gym/gymnasium vectorized envs or Brax, for example) autoresets sub-environments after they terminate or truncated.
This is a difference with respect to frameworks that only implement a single environment (such as gym/gymnasium non-vectorized (de default) or DeepMind, for example) where it is necessary to invoke the method to reset the environments “manually”.

Now, `skrl` is a generic RL library that not only support Isaac Lab but other RL frameworks, and is therefore able to handle both cases.

For Isaac Lab (and other frameworks that support parallel/vectorized envs), the `.reset()` method only "reset" the environment once. Subsequent calls just return the latest `.step()` observation/info as referenced in the next code:

https://github.com/Toni-SM/skrl/blob/441910440ffc53fa88256d0a9f34d60922c1eac5/skrl/envs/wrappers/torch/isaaclab_envs.py#L66-L75

Moreover, the sequential trainer (designed to train more than one agent simultaneously in parallel environments) calls the implementation of the base class (when there are no simultaneous agents to train, current Isaac Lab case). The base class implementation takes into account whether the environment is parallel/vectorized or not, to avoid calling the reset method (even if it is safe to invoke it) as referenced in the next code:

https://github.com/Toni-SM/skrl/blob/441910440ffc53fa88256d0a9f34d60922c1eac5/skrl/trainers/torch/base.py#L208-L216

<hr>

Therefore, I strongly recommend that you continue to use `skrl` (and with it the improved implementation of the [Model instantiators' Network definitions](https://skrl.readthedocs.io/en/latest/api/utils/model_instantiators.html#network-definitions) currently available in skrl-v1.3.0 and integrated in the upcoming official version of Isaac Lab).
